analytics/src/analytics/etl/github.py

```diff
@@ -0,0 +1,312 @@
+"""Extract and transform GitHub data to create the `GitHubIssues` dataset."""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+
+import pandas as pd
+from pydantic import BaseModel, ValidationError
+
+from analytics.datasets.issues import (
+    GitHubIssues,
+    IssueMetadata,
+    IssueType,
+)
+from analytics.datasets.utils import load_json_file
+from analytics.integrations import github
+
+logger = logging.getLogger(__name__)
+
+# ========================================================
+# Config interfaces
+# ========================================================
+
+
+@dataclass
+class InputFiles:
+    """Expected input files for loading this dataset."""
+
+    roadmap: str
+    sprint: str
+
+
+class GitHubProjectConfig(BaseModel):
+    """Configurations for GitHub projects ETL."""
+
+    roadmap_project: RoadmapConfig
+    sprint_projects: list[SprintBoardConfig]
+    temp_dir: str = "data"
+    output_file: str = "data/delivery-data.json"
+
+
+class RoadmapConfig(BaseModel):
+    """
+    Configuration details for a roadmap project.
+
+    Attributes
+    ----------
+    owner
+        The GitHub login for the project owner.
+    project_number
+        The number of the project.
+    quad_field
+        The name of the project field that stores the Quad value for a deliverable.
+    pillar_field
+        The name of the project field that stores the Pillar value for a deliverable.
+    """
+
+    owner: str
+    project_number: int
+    quad_field: str = "Quad"
+    pillar_field: str = "Pillar"
+
+
+class SprintBoardConfig(BaseModel):
+    """
+    Configuration details for a sprint board project.
+
+    Attributes
+    ----------
+    owner
+        The GitHub login for the project owner.
+    number
+        The number of the project.
+    points_field
+        The name of the project field that stores story points or estimates.
+    sprint_field
+        The name of the project field that stores the sprint value.
+    """
+
+    owner: str
+    project_number: int
+    points_field: str = "Points"
+    sprint_field: str = "Sprint"
+
+
+# ========================================================
+# ETL pipeline
+# ========================================================
```

Comment:
All of the code below this was relocated from `/analytics/datasets/issues.py` after creating this new ETL sub-package